### PR TITLE
client/web: move auth session creation out of /api/auth

### PR DESCRIPTION
**Pull request by soniaappasamy**
Splits auth session creation into two new endpoints:

/api/auth/session/new - to request a new auth session

/api/auth/session/wait - to block until user has completed auth url

Updates tailscale/corp#14335

### CODE REVIEW

File: client/web/web.go [hunk 1]
```diff
@@ -328,12 +330,47 @@ func (s *Server) serveAPIAuth(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+}
+
+type newSessionAuthResponse struct {
```
[the remark this answers is not in the document]
Comment (soniaappasamy) [reply client/web/web.go:357]: Kinda like keeping them separate? Since we no longer need the URL in `authResponse`

**Comment by soniaappasamy** [comment]
Updated w/ web_test.go fixes

File: client/web/web.go [hunk 2]
```diff
@@ -328,12 +330,47 @@ func (s *Server) serveAPIAuth(w http.ResponseWriter, r *http.Request) {
...
+		resp = authResponse{OK: true}
+	default:
+		resp = authResponse{OK: false, AuthNeeded: tailscaleAuth}
+	}
+
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
```
[the remark this answers is not in the document]
Comment (soniaappasamy) [reply client/web/web.go]: added a `writeJson` func to help with this

[the remark this answers is not in the document]
Comment (bradfitz) [reply client/web/web.go]: writeJSON per https://github.com/golang/go/wiki/CodeReviewComments#initialisms

[the remark this answers is not in the document]
Comment (soniaappasamy) [reply client/web/web.go]: Hahaha actually I did name it that in the code, I just commented the wrong thing here

**Review by willnorris — APPROVED** [review]

**Pull request merged**